atlasexec/atlas.go

```diff
@@ -188,9 +188,9 @@ func NewClient(workingDir, execPath string) (_ *Client, err error) {
 //	  })
 //	  return err
 //	})
-func (c *Client) WithWorkDir(dir string, fn func(*Client) error) error {
+func (c Client) WithWorkDir(dir string, fn func(*Client) error) error {
 	c.workingDir = dir
-	return fn(c)
+	return fn(&c)
 }
```

Comment:
why not this way?

```
func (c *Client) WithWorkDir(dir string, fn func(*Client) error) error {
	wd, c.workingDir =  c.workingDir, dir
	err := fn(c)
	c.workingDir =  wd
	return err
}
```

Reply:
😄 okay

Reply:
Changed. But I use `defer` to avoid defining the `wd` var at the beginning.

Reply:
> Changed. But I use `defer` to avoid defining the `wd` var at the beginning.

You still need to define it.